bloom-housing/bloom
# feat: add application confirmation modals

This PR addresses #4047


## Description

Adds two different confirmation modals on the main applications table page in partners that requires users to confirm that they would like to add an application if
(1) The application is closed
(2) A lottery has already been run

Confirmed that the first modal should be present regardless of whether or not lotteries is toggled on, and to separate the copy in the lottery modal to two sentences.

## How Can This Be Tested/Reviewed?

Visit an open listing and click the add application button and ensure you go straight to the application form.
Visit a closed closed and click the add application button and ensure the closed listing modal from the [design](https://www.figma.com/design/PAeCVANDmGdTz2Y53ouErM/Lotteries?node-id=3235-37543&t=BUWB2SeiXVzj60ex-0) is shown.
There is a new backend field `lotteryLastRunAt` that is not currently used by the backend. If that field is not null, the lottery run modal from the [design](https://www.figma.com/design/PAeCVANDmGdTz2Y53ouErM/Lotteries?node-id=3235-28349&t=P7k2YlCnvTwanwbW-0) is shown. You can either set this field on a listing, or just manually toggle that if statement on to test it out.

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
